### PR TITLE
Added Backtracking algorithm (N-Queen Problem) [Go]

### LINKED ISSUE #535: Backtracking algorithm (N-Queen Problem) [Go]
#### This issue is about [Backtracking algorithm (N-Queen Problem)](https://en.wikipedia.org/wiki/Eight_queens_puzzle).

- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [x] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by vatz88**
Fixes #535

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

File: n_queen_problem/n_queen_problem.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+// This implementation prints only one valid solution, it can be extended to print all possible valid solutions.
+
+package main
+
+import "fmt"
+
+func main() {
+	var n int
+	print("Enter the value of 'N' ")
+	fmt.Scanf("%d", &n)
```
Comment (mohitkyadav) [inline n_queen_problem/n_queen_problem.go]: No need to ask the value of `n`, take any sample value.

Comment (vatz88) [reply n_queen_problem/n_queen_problem.go]: Ok I'll set it to 8

File: n_queen_problem/n_queen_problem.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+				if (*b)[i][j] == 1 {
+					return true
+				}
+			}
+		}
+	}
+	return false
+}
+
+func placeQueen(b *[][]int, n int) bool {
```
Comment (mohitkyadav) [inline n_queen_problem/n_queen_problem.go]: Please export this func.

File: n_queen_problem/n_queen_problem.go [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+package main
+
+import "fmt"
+
+func main() {
+	n := 8 // size of chess board (n x n)
+
+	// make board
+	b := make([][]int, n)
```
Comment (mohitkyadav) [inline n_queen_problem/n_queen_problem.go]: Please use common variable names such as, n => size, b => board.

Comment (vatz88) [reply n_queen_problem/n_queen_problem.go]: Yes that makes more sense. Done.

File: n_queen_problem/n_queen_problem.go (hunk 2 above)
Comment (vatz88) [reply n_queen_problem/n_queen_problem.go]: I don't know what that exactly means but changed the function name to `PlaceQueen` and added a comment before it as explanation.

**Pull request merged**